File: GeoServices/SDE/TableGateway.vb
Imports ESRI.ArcGIS.Geodatabase

Namespace SDE
    Public Class TableGateway
        Inherits SDEDataGateway(Of ITable)

        ''' <summary>
        ''' Obtiene todas las tablas del SDE para la conexión especificada
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks>El esquema de obtención está realizado para tablas sin anidamiento que se encuentran en la raíz del SDE</remarks>
        Protected Overrides Function doGetAll(ByVal workspace As IWorkspace, ByVal RequiresEditorPriviledges As Boolean) As System.Collections.Generic.List(Of ESRI.ArcGIS.Geodatabase.ITable)
            Dim tables As New List(Of ITable)

            Dim datasets As IEnumDataset = workspace.Datasets(esriDatasetType.esriDTTable)
            Dim table As IDataset = datasets.Next
            While Not table Is Nothing
                If Me.ExtraValidation(table, RequiresEditorPriviledges) Then tables.Add(table)
                table = datasets.Next
            End While

            System.Runtime.InteropServices.Marshal.ReleaseComObject(datasets)

            Return tables
        End Function

        Protected Overrides Function GetElementName() As String
            Return "Tabla"
        End Function

        Protected Overrides Function GetPluralName() As String
            Return "Tablas"
        End Function

        Protected Overrides Function IsNameEquals(ByVal element As ESRI.ArcGIS.Geodatabase.ITable, ByVal name As String) As Boolean
            Return CType(element, IDataset).Name.ToUpper().Contains(name.ToUpper())
        End Function
    End Class
End Namespace

File: GeoServicesSharp/SDE/TableGateway.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ESRI.ArcGIS.Geodatabase;

namespace GeoServicesSharp.SDE
{
    public class TableGateway : SDEDataGateway<ITable>
    {
        /// <summary>
        /// Obtiene todas las tablas del SDE para la conexión especificada
        /// </summary>
        /// <remarks>El esquema de obtención está realizado para tablas sin anidamiento que se encuentran en la raíz del SDE</remarks>
        protected override List<ITable> doGetAll(IWorkspace workspace, SDEPrivileges Privileges)
        {
            List<ITable> tables = new List<ITable>();

            IEnumDataset datasets = workspace.get_Datasets(esriDatasetType.esriDTTable);
            IDataset table = datasets.Next();
            while (table != null)
            {
                if (this.PermissionsValidation((ITable)table, Privileges)) tables.Add((ITable)table);
                table = datasets.Next();
            }

            System.Runtime.InteropServices.Marshal.ReleaseComObject(datasets);

            return tables;
        }

        protected override bool IsNameEquals(ITable element, string name)
        {
            return ((IDataset)element).Name.ToUpper().Contains(name.ToUpper());
        }

        protected override ITable doGetByName(string name, IFeatureWorkspace workspace)
        {
            return workspace.OpenTable(name);
        }
    }
}
